docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,383 @@
+        "restaurant": [
+            {
+                "id": 0,
+                "name": "Donath",
+                "cuisine": "Italian",
+                "outside-seating": true,
+                "price-range": "mid-range"
+            },
+            {
+                "id": 1,
+                "name": "Berlin Burrito Company",
+                "cuisine": "Mexican",
+                "outside-seating": false,
+                "price-range": "cheap"
+            },
+            {
+                "id": 2,
+                "name": "I due forni",
+                "cuisine": "Italian",
+                "outside-seating": true,
+                "price-range": "mid-range"
+            }
+        ],
+        "hotel": [
+            {
+                "id": 0,
+                "name": "Hilton",
+                "price-range": "expensive",
+                "breakfast-included": true,
+                "city": "Berlin",
+                "free-wifi": true,
+                "star-rating": 5,
+                "swimming-pool": true
+            },
+            {
+                "id": 1,
+                "name": "Hilton",
+                "price-range": "expensive",
+                "breakfast-included": true,
+                "city": "Frankfurt am Main",
+                "free-wifi": true,
+                "star-rating": 4,
+                "swimming-pool": false
+            },
+            {
+                "id": 2,
+                "name": "B&B",
+                "price-range": "mid-range",
+                "breakfast-included": false,
+                "city": "Berlin",
+                "free-wifi": false,
+                "star-rating": 1,
+                "swimming-pool": false
+            },
+        ]
+    }
+
+The above json file contains data about restaurants and hotels.
+The json structure should contain a key for every object type, i.e. "restaurant" and "hotel".
+Every object type maps to a list of objects.
+
+Once the data are defined in a json file, called, for example, ``data.json``, you can create your
+``InMemoryKnowledgeBase``:
+
+.. code-block:: python
+
+    knowledge_base = InMemoryKnowledgeBase.load("data.json")
+
+Customize your InMemoryKnowledgeBase
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The class ``InMemoryKnowledgeBase`` inherits ``KnowledgeBase``.
+You can customize your ``InMemoryKnowledgeBase`` by overwriting the following functions:
+
+- ``get_key_attribute_of_object``: To keep track of what object the user was talking about last, we store the value
+  of the key attribute in a specific slot. Every object should have a key attribute that is unique, i.e.
+  similar to the primary key in a relation database. Per default the name of the key attribute for every object type
+  is set to "id". You can overwrite the name of the key attribute for a specific object type by calling
+  ``set_key_attribute_of_object()``.
+- ``get_representation_function_of_object``: This methods returns a lambda function that maps an object from the
```

Comment:
this is a difficult sentence (part of a difficult paragraph). Let's split it up into smaller sentences and give quick examples